Khan/wonder-blocks
# move the backdrop inside the focus trap so that focus in dropdowns works properly

The FocusTrap sets up "sentinel" elements in the DOM and works to guarantee that focus always stays within the sentinels.  Unfortunately, we were mounting portals for things like tooltips and dropdowns that live inside modals, outside of these sentinels.  The result is that clicking an item in a dropdown would immediately focus the close button.  This PR fixes that but moving the whole ModalBackdrop inside the FocusTrap (and inside the sentinels).  This will also come in handy when we get around to adding the prev/next bumpers for navigating between content modals.  This also makes the FocusTrap easier to reuse b/c now it's responsible for setting up the sentinels.

Test Plan:
- load http://localhost:6060/#!/MultiSelect
- open the modal
- open the dropdown
- select an item, see that the close button isn't focused
- click on the backdrop to dismiss the modal
- use the keyboard to reopen the modal
- press space to open the dropdown
- tab through the items and select some of them with space
- tab though all of the items and keep pressing tab until you get back to dropdown, at no point should focus leave the modal or its contents

Unfortunately the FocusTrap can't be tested with enzyme + jest.  We'll need to e2e tests using vite for that.
